crates/anvil-polkadot/tests/it/txpool.rs

```diff
@@ -0,0 +1,140 @@
+    let alith_addr = Address::from(ReviveAddress::new(alith.address()));
+    let recipient_addr = Address::repeat_byte(0x42);
+
+    let balance = node.get_balance(alith.address(), None).await;
+    assert!(balance > U256::ZERO, "Alith should have balance");
+
+    unwrap_response::<()>(node.eth_rpc(EthRequest::SetAutomine(false)).await.unwrap()).unwrap();
+
+    // Send 2 transactions with nonce 0 and 1
+    let tx1 = TransactionRequest::default()
+        .from(alith_addr)
+        .to(recipient_addr)
+        .value(U256::from(1000))
+        .nonce(0);
+    node.send_transaction(tx1, None).await.unwrap();
+
+    let tx2 = TransactionRequest::default()
+        .from(alith_addr)
+        .to(recipient_addr)
+        .value(U256::from(2000))
+        .nonce(1);
+    let tx2_hash = node.send_transaction(tx2, None).await.unwrap();
+
+    // Verify pool has 2 pending transactions
+    let status: TxpoolStatus =
+        unwrap_response(node.eth_rpc(EthRequest::TxPoolStatus(())).await.unwrap()).unwrap();
+    assert_eq!(status.pending, 2, "Pool should have 2 pending transactions");
+
+    // Drop transaction with nonce 1
+    let tx2_hash_b256 = B256::from_slice(tx2_hash.0.as_ref());
+    unwrap_response::<()>(node.eth_rpc(EthRequest::DropTransaction(tx2_hash_b256)).await.unwrap())
+        .unwrap();
+
+    // Verify only transaction with nonce 0 remains
+    let status: TxpoolStatus =
+        unwrap_response(node.eth_rpc(EthRequest::TxPoolStatus(())).await.unwrap()).unwrap();
+    assert_eq!(status.pending, 1, "Pool should have 1 pending transaction after drop");
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_drop_all_transactions() {
+    let anvil_node_config = AnvilNodeConfig::test_config();
+    let substrate_node_config = SubstrateNodeConfig::new(&anvil_node_config);
+    let mut node = TestNode::new(anvil_node_config, substrate_node_config).await.unwrap();
+
+    unwrap_response::<()>(node.eth_rpc(EthRequest::SetAutomine(true)).await.unwrap()).unwrap();
+
+    let alith = Account::from(subxt_signer::eth::dev::alith());
+    let alith_addr = Address::from(ReviveAddress::new(alith.address()));
+    let recipient_addr = Address::repeat_byte(0x42);
+
+    let balance = node.get_balance(alith.address(), None).await;
+    assert!(balance > U256::ZERO, "Alith should have balance");
+
+    unwrap_response::<()>(node.eth_rpc(EthRequest::SetAutomine(false)).await.unwrap()).unwrap();
+
+    // Send 3 transactions
+    for i in 0..3 {
+        let tx = TransactionRequest::default()
+            .from(alith_addr)
+            .to(recipient_addr)
+            .value(U256::from(1000 * (i + 1)))
+            .nonce(i);
+        node.send_transaction(tx, None).await.unwrap();
+    }
+
+    // Verify pool has 3 pending transactions
+    let status: TxpoolStatus =
+        unwrap_response(node.eth_rpc(EthRequest::TxPoolStatus(())).await.unwrap()).unwrap();
+    assert_eq!(status.pending, 3, "Pool should have 3 pending transactions");
```

Comment:
Again, I would expect transaction with nonce 2 to be queued, and to be declared future by txpool, so at least one of the txs should make their way to the `queued` set. But maybe if a sequence of consecutive nonced txs are sent to the node they make their way in order to the ready set. Need to double check the txpool code for this one.

Reply:
> Need to double check the txpool code for this one.

I actually asked our txpool authority (Michal), haven't checked yet the code, and the answer is that they should be in the ready set. Maybe we can also send a tx to end up in the `queued` set, so that dropping all will rightly show both `queued` and `pending` sets empty.